spring-integration-debezium/src/main/java/org/springframework/integration/debezium/dsl/DebeziumMessageProducerSpec.java

```diff
@@ -0,0 +1,134 @@
+	 * serializable. By default, the batch mode is disabled, e.g. every input {@link ChangeEvent} is converted into a
+	 * single downstream {@link Message}.
+	 * @param enable set to true to enable the batch mode. Disabled by default.
+	 * @return the spec.
+	 */
+	public DebeziumMessageProducerSpec enableBatch(boolean enable) {
+		this.target.setEnableBatch(enable);
+		return this;
+	}
+
+	/**
+	 * Enable support for tombstone (aka delete) messages. On a database row delete, Debezium can send a tombstone
+	 * change event that has the same key as the deleted row and a value of {@link Optional#empty()}. This record is a
+	 * marker for downstream processors. It indicates that log compaction can remove all records that have this key.
+	 * When the tombstone functionality is enabled in the Debezium connector configuration you should enable the empty
+	 * payload as well.
+	 * @param enabled set true to enable the empty payload. Disabled by default.
+	 * @return the spec.
+	 */
+	public DebeziumMessageProducerSpec enableEmptyPayload(boolean enabled) {
+		this.target.setEnableEmptyPayload(enabled);
+		return this;
+	}
+
+	/**
+	 * Set a {@link ThreadFactory} for the Debezium executor. Defaults to the {@link CustomizableThreadFactory} with a
+	 * {@code debezium:inbound-channel-adapter-thread-} prefix.
+	 * @param threadFactory the {@link ThreadFactory} instance to use.
+	 * @return the spec.
+	 */
+	public DebeziumMessageProducerSpec threadFactory(ThreadFactory threadFactory) {
+		this.target.setThreadFactory(threadFactory);
+		return this;
+	}
+
+	/**
+	 * Set the outbound message content type. Must be aligned with the {@link SerializationFormat} configuration used by
+	 * the provided {@link DebeziumEngine}.
+	 * @param contentType payload content type.
+	 * @return the spec.
+	 */
+	public DebeziumMessageProducerSpec contentType(String contentType) {
+		this.target.setContentType(contentType);
+		return this;
+	}
+
+	/**
+	 * Comma-separated list of names of {@link ChangeEvent} headers to be mapped into outbound Message headers.
+	 * Debezium's NewRecordStateExtraction 'add.headers' property configures the metadata to be used as
+	 * {@link ChangeEvent} headers.
+	 * <p>
+	 * You should prefix the names passed to the 'headerNames' with the prefix configured by the Debezium
+	 * 'add.headers.prefix' property. Later defaults to '__'. For example for 'add.headers=op,name' and
+	 * 'add.headers.prefix=__' you should use header hames like: '__op', '__name'.
+	 * @param headerNames The values in this list can be a simple patterns to be matched against the header names.
+	 * @return the spec.
+	 */
+	public DebeziumMessageProducerSpec headerNames(String... headerNames) {
+		Assert.notEmpty(headerNames, "'inboundHeaderNames' must not be empty");
+		this.inboundHeaderNames = Arrays.copyOf(headerNames, headerNames.length);
+		return addDebeziumHeaderMapper();
```

Comment:
I don't see a reason in this method: you simply can create a `DefaultDebeziumHeaderMapper`, initialize it here and then call `headerMapper()`. 
Also must be mentioned that this `headerNames()` and `headerMapper()` are mutually exclusive.

And from here we might not need that `inboundHeaderNames` property at all.